core/app.cc

```diff
@@ -401,6 +402,14 @@ void application::prepare_argv(elf::program *program)
     _auxv[auxv_idx].a_type = AT_PAGESZ;
     _auxv[auxv_idx++].a_un.a_val = sysconf(_SC_PAGESIZE);
 
+    _auxv[auxv_idx].a_type = AT_MINSIGSTKSZ;
+    _auxv[auxv_idx++].a_un.a_val = sysconf(_SC_MINSIGSTKSZ);
+
+    random_bytes[0] = rand();
+    random_bytes[1] = rand();
```

Comment:
I have no idea what's the purpose of these random bytes, but calling rand() might not be the right thing to do... It's not thread-safe (although I guess the risk in practice is near zero) and uses some seed that the user might have set for a different purpose. Using your own seed (like the current time) and then rand_r() is probably better, although I wonder if you need to get higher quality random numbers from /dev/urandom or something?

More seriously, rand() only returns an "int", i.e., 4 random bytes, not 8. So half of your "random bytes" are zero, unless I'm missing something.

Reply:
I think a better solution is to use `getrandom()` and if it fails fall back to `rand_r()`.